packages/xchain-ethereum/src/types/client-types.ts

```diff
@@ -18,40 +18,21 @@ export type ExplorerUrl = {
   mainnet: string
 }
 
-export type VaultTxOpts = {
-  address: Address
-  amount: BaseAmount
-  memo: string
-  overrides?: ContractTxOverrides
-}
-
-export type NormalTxOpts = {
+export type EstimateGasOpts = {
```

Comment:
I think you might need to add an optional `from` if you'd like to query the gas estimate depending of the sender; without having him to pass his private key everytime;

The `from` is important because if the recipient is a smart contract that behaves differently depending of the `sender`, his account balance (and the smart contract state), it will cost different gas.

Reply:
On Contract instance:

```
// https://docs.ethers.io/v5/api/contract/contract/#contract-estimateGas
overrides = { from: '0x....' }
contract.estimateGas.METHOD_NAME( ...args [ , overrides ] )
```

Reply:
Thank you!